src/simple-keyring.ts

```diff
@@ -0,0 +1,226 @@
+import { TxData, TypedTransaction } from '@ethereumjs/tx';
+import {
+  arrToBufArr,
+  bufferToHex,
+  ecsign,
+  isValidPrivate,
+  privateToPublic,
+  publicToAddress,
+  stripHexPrefix,
+  toBuffer,
+} from '@ethereumjs/util';
+import {
+  concatSig,
+  decrypt,
+  getEncryptionPublicKey,
+  normalize,
+  personalSign,
+  signTypedData,
+  SignTypedDataVersion,
+} from '@metamask/eth-sig-util';
+import { add0x, Eip1024EncryptedData, Hex, Keyring } from '@metamask/utils';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import randombytes from 'randombytes';
+
+const _TYPE = 'Simple Key Pair';
+
+export default class SimpleKeyring implements Keyring<string[]> {
+  #wallets: { privateKey: Buffer; publicKey: Buffer }[];
+
+  readonly type: string = _TYPE;
+
+  static type: string = _TYPE;
+
+  constructor(options: string[] = []) {
+    this.#wallets = [];
+
+    // istanbul ignore next
```

Comment:
I actually tried pretty hard on this one (on March 19, before you wrote this).  I got stuck because a constructor isn't allowed to be async (so technically the fact that it calls an async method is weird).  Jest can't await the constructor, and when the error gets thrown, jest can't catch it.  Below is one example of a test that does not work.

```
  describe('#constructor with an invalid key', function () {
    it('throws an error', async function () {
      try {
        const newKeyring = new SimpleKeyring(['invalid_key']);
      } catch (e) {
        console.log('caught the error', e);
      }
    });
  });
  ```

Reply:
Oh, sure. That makes sense. Maybe it'd be good to add a comment above `// istanbul ignore next` explaining why we are not testing this then? We typically do the same thing for `// eslint-ignore` lines too.